src/test/ui/consts/const_in_pattern/reject_non_partial_eq.stderr

```diff
@@ -1,4 +1,4 @@
-error: to use a constant of type `NoPartialEq` in a pattern, `NoPartialEq` must be annotated with `#[derive(PartialEq, Eq)]`
+error: to use a constant of type `Option<NoPartialEq>` in a pattern, `Option<NoPartialEq>` must be annotated with `#[derive(PartialEq, Eq)]`
```

Comment:
Hmm this looks potentially like a downgrade, is this because you check "structural eq" deeply on the `Option<NoPartialEq>`, as opposed to shallowly?